### ISSUE TITLE
Freeze when pasting

### ISSUE THREAD

**Issue by afranke** [aperezdc/revolt#54]
Since I updated Revolt today it freezes every time I dare use ctrl + v to paste something.

**Comment by nielsdg** [comment]
Can confirm this bug.

**Comment by aperezdc** [comment]
@afranke @nielsdg: Are you using a X11 or a Wayland session? Paste seems fine under X11 for me. Under Wayland sometimes I have also seen that sometimes using `Ctrl+V` will (seemingly) freeze Revolt, but then if I go to another application an press `Ctrl+C`, the paste finishes and Revolt continues working fine. For me it seems to happen when:

- Revolt is being run in a Wayland session.
- The pasted content comes from the X11 pasteboard (some application running under XWayland).

Is this case I am experiencing the same as for you? Or does pressing `Ctrl+V` always freeze the application for you? Some details on which kind of session are you using, and whether the source program from where content was copied to the clipboard would be handy (and in particular whether you are using Wayland/X11/XWayland).

**Comment by afranke** [comment]
Wayland session. I can confirm that if I copy from somewhere then:

1. ctrl + v in Revolt, Revolt freezes
2. switch to gedit, ctrl + v in gedit, switch back to Revolt, Revolt is not frozen anymore.

As I said in comment 1 this was not an issue before, only since the update.

It happens when copying from Firefox (which I assume is not using Wayland yet) but not from gedit.

**Comment by aperezdc** [comment]
@afranke Indeed, Firefox still does not use Wayland (there are unofficial [Flatpak'd builds of Firefox for Wayland](https://firefox-flatpak.mojefedora.cz/), so it's definitely going to be supported on Wayland at some point!).

I think we are hitting [this GTK+ bug](https://bugzilla.gnome.org/show_bug.cgi?id=781814), so I probably the next release will make the clipboard working reliably in this case. Let's keep this issue open and we can close it later on.
